packages/engine/lib/stateful/src/agent/arrow/batch.rs

```diff
@@ -35,10 +35,16 @@ use crate::{
     field::{POSITION_DIM, UUID_V4_LEN},
 };
 
-/// An Arrow batch with agent state columns
-// TODO: Maybe rename to AgentGroup
+/// A list of [`Agent`] fields.
+///
+/// Internally, this is backed by an [`ArrowBatch`], which is organizing the agent fields as a
+/// struct of arrays.
+// TODO: Maybe come up with a better name. Actually, this is not a batch of `Agent`s as the name
+//   suggested but only the agent fields, so this is not a group of agents but only a collection of
+//   agent fields
```

Comment:
Not really sure about this TODO. The TODO itself seems to be confusing Groups with Batches. Nothing about the name mentions groups, and groups **are not** batches.

I think `AgentBatch` is fine, as we're literally just wrapping `ArrowBatch` and adding `worker_index` metadata.

Unless I'm misunderstanding the TODO and it's instead saying that calling it "`Agent`" is confusing. In which case I don't mind if we end up swapping this to be `AgentStateBatch` or something.

Reply:
I think renaming `AgentBatch` to `AgentStateBatch` is good to avoid confusion. I removed the TODO for now and did the change. Any idea where we could describe the concept of groups vs agent batch we previously talked about?

Reply:
> I think `AgentBatch` is fine, as we're literally just wrapping `ArrowBatch` and adding `worker_index` metadata.

I'm unsure if this is the right place to put `worker_index` generally. We storing information in the struct which is unrelated to this crate.